File: packages/plugin-contract/src/severity.ts
/**
 * Clasificación de severidad de una acción (ADR-004, docs/00).
 * Determina si el Permission Manager la ejecuta directo o exige confirmación
 * explícita antes de tocar el dispositivo real.
 */
export type ActionSeverity =
  | "read-only"
  | "reversible"
  | "irreversible-material"
  | "safety-critical";

export function requiresConfirmation(severity: ActionSeverity): boolean {
  return severity === "irreversible-material" || severity === "safety-critical";
}

File: packages/kan-plugin-sdk-py/src/kan_plugin_sdk_py/severity.py
"""Espejo a mano de packages/plugin-contract/src/severity.ts (ADR-004, docs/00)."""

from typing import Literal

ActionSeverity = Literal["read-only", "reversible", "irreversible-material", "safety-critical"]


def requires_confirmation(severity: ActionSeverity) -> bool:
    return severity in ("irreversible-material", "safety-critical")
